### PR TITLE
Align docker image for backend with current best practices

### LINKED ISSUE #986: Align backend Dockerfiles with the current security best practices
There's been a very nice [blog post by Snyk](https://snyk.io/blog/10-best-practices-to-containerize-nodejs-web-applications-with-docker/) listing some best practices for Dockerfiles. Currently, Titus backend is not using most of them. It would be useful to have the best practices applied by default, as usually, security is an afterthought. I don't think making those changes would get in anyone's way.

### PR DESCRIPTION
**Pull request by Vunovati**
Adjust the Dockerfile according to tips in this article:
https://snyk.io/blog/10-best-practices-to-containerize-nodejs-web-applications-with-docker/

Resolve #986

### CODE REVIEW

**Review by leorossi — APPROVED** [review]
LGTM

File: packages/titus-backend/Dockerfile [hunk 1]
```diff
@@ -1,13 +1,15 @@
-FROM node:14-alpine as builder
+FROM node:lts-alpine@sha256:fb6cb918cc72869bd625940f42a7d8ae035c4e786d08187b94e8b91c6a534dfd as builder
```
Comment (simoneb) [inline packages/titus-backend/Dockerfile:1]: this is the only one part that I don't necessarily agree with. as this is an accelerator, it is supposed to be future proof. using whatever node 14 lts is the latest is a good way to ensure that you get... the latest 14 lts, which is stable enough and especially may contain security fixes which are not in the revision that you pinned. let's restore it to 14-alpine

**Review by simoneb — COMMENTED** [review]
Overall LGTM, one comment about the image tag.

Please make sure to test it locally and, if all works fine, follow the CD deployment to GCP once merged to make sure it keeps running fine on GCP

File: packages/titus-backend/Dockerfile (hunk 1 above)
Comment (Vunovati) [reply packages/titus-backend/Dockerfile]: Makes sense, this is an accelerator, not a final product.

**Review by simoneb — APPROVED** [review]
LGTM

**Pull request merged**